utils/navigator_robotx_comms/nodes/navigator_robotx_comms.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+
+"""
+RobotX Communications: A node that handles communications
+with the RobotX Technical Director network.
+"""
+import errno
+
+import rospy
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import PointStamped
+import socket
+import datetime
+import time
+
+rospy.init_node("robotx_comms")
+
+
+class RobotXComms():
+    """
+    Handles communications
+    with the RobotX Technical Director network.
+    """
+    system_mode = None  # type: int
+
+    def __init__(self):
+
+        self.gps_array = None
+        self.odom = None
+        self.auv_status = 1
+        self.system_mode = 1
+        # self.TEAM_ID = rospy.get_param(team_id)
+        self.team_id = "GATOR"
+
+        # rospy.Subscriber("auv_status", String, self.auv_status_callback)
+
+    def heartbeat_msg(self):
+
+        message_id = "$RXHRB"
+        delim = ","
+
+        # HST is 10 hours behind UTC
+        hst_time = datetime.datetime.utcnow() - datetime.timedelta(hours=10, minutes=0)
+        date_string = hst_time.strftime("%d%m%y")
+        time_string = hst_time.strftime("%H%M%S")
+
+        if self.gps_array is not None:
+            latitude = self.gps_array.point.x
+            longitude = self.gps_array.point.y
+        else:
+            latitude = ""
+            longitude = ""
+
+        if self.odom is not None:
+            quaternion = self.odom.pose.pose.orientation
+            north_south = ""
+            east_west = ""
+            if quaternion.x > 0:
+                east_west = "E"
+            elif quaternion.x < 0:
+                east_west = "W"
+            if quaternion.y > 0:
+                north_south = "N"
+            elif quaternion.y < 0:
```

Comment:
We talked about this at lakeday, but remember to update this so use the yaw angle, which is much easier to reason about for something like this cardinal direction identification